processor/logdeduplicationprocessor/config.go

```diff
@@ -0,0 +1,76 @@
+// Copyright  observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package logdeduplicationprocessor provides a processor that counts logs as metrics.
+package logdeduplicationprocessor
+
+import (
+	"errors"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+)
+
+// Config defaults
+const (
+	// defaultInterval is the default export interval.
+	defaultInterval = 10 * time.Second
+
+	// defaultLogCountAttribute is the default log count attribute
+	defaultLogCountAttribute = "log_count"
+
+	// defaultTimezone is the default timezone
+	defaultTimezone = "UTC"
+)
+
+// Config errors
+var (
+	errInvalidLogCountAttribute = errors.New("log_count_attribute must be set")
+	errInvalidInterval          = errors.New("interval must be greater than 0")
+	errInvalidTimezone          = errors.New("timezone is invalid")
+)
+
+// Config is the config of the processor.
+type Config struct {
+	LogCountAttribute string        `mapstructure:"log_count_attribute"`
+	Interval          time.Duration `mapstructure:"interval"`
+	Timezone          string        `mapstructure:"timezone"`
+}
+
+// createDefaultConfig returns the default config for the processor.
+func createDefaultConfig() component.Config {
+	return &Config{
+		LogCountAttribute: defaultLogCountAttribute,
+		Interval:          defaultInterval,
+		Timezone:          defaultTimezone,
+	}
+}
+
+// Validate validates the configuration
+func (c Config) Validate() error {
+	if c.Interval <= 0 {
+		return errInvalidInterval
+	}
+
+	if c.LogCountAttribute == "" {
+		return errInvalidLogCountAttribute
+	}
+
+	_, err := time.LoadLocation(c.Timezone)
+	if err != nil {
+		return errInvalidTimezone
+	}
```

Comment:
I think it would be useful to wrap the error from LoadLocation here, in case the error is more than just a bad timezone string.

Reply:
I'm debating this. My original thought was we point to a list of acceptable values in documentation. I don't want to really expose that they can do stuff outside of that list and directly expose the user to `time.LoadLocation`.

Reply:
I think the problem is that there's other stuff that can go wrong with time.LoadLocation based on environment; Like, if the ZONEINFO env variable is present, for instance. So there actual error could be useful.

Reply:
Oh right we've hit this before too. I'll add it.